src/test/java/com/puppycrawl/tools/checkstyle/xpath/AttributeNodeTest.java

```diff
@@ -49,6 +50,27 @@ public void testCompareOrder() {
         }
     }
 
+    @Test
+    public void testGetDepth() {
+        try {
+            attributeNode.getDepth();
+            fail("Exception is excepted");
+        }
+        catch (UnsupportedOperationException ex) {
```

Comment:
we need to think on what approach for exception validation is better for us
https://www.baeldung.com/junit-assert-exception

usage is Rule is good, junit5 approach also looks interesting but still creates nesting blocks.

@strkkk , @rnveach , @nmancus1 , what is more preferable way of exception validation for you.

Reply:
Sounds good. I'll try to use assertThrows + Truth.

Reply:
done

Reply:
`assertThrows` looks like a better way to me

Reply:
I would prefer https://assertj.github.io/doc/ over either JUnit4 or JUnit5 exception validation (read more at https://www.baeldung.com/introduction-to-assertj), but I am not sure if this is an option that we could use. In my opinion, it is the most readable(by far) of the three. 

Out of the two options given above `@Rule` (JUnit4) and `assertThrows` (JUnit5), *I think that `@Rule` is more readable*. However, JUnit5 is overall a better UT framework, since we can import only the modules we need, and can run tests in parallel.

I would rank the three options as follows:
1. assertJ
2. JUnit5 `assertThrows`
3. JUnit4 `@Rule`